pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -360,33 +361,72 @@ func (ca *cloudCIDRAllocator) updateCIDRAllocation(nodeName string) error {
 			//
 			// See https://github.com/kubernetes/kubernetes/pull/42147#discussion_r103357248
 		}
-		for i := 0; i < cidrUpdateRetries; i++ {
-			if err = utilnode.PatchNodeCIDRs(ca.client, types.NodeName(node.Name), cidrStrings); err == nil {
-				klog.InfoS("Set the node PodCIDRs", "nodeName", node.Name, "cidrStrings", cidrStrings)
-				break
-			}
-		}
+		node.Spec.PodCIDR = cidrStrings[0]
+		node.Spec.PodCIDRs = cidrStrings
 	}
+
+	err = ca.updateNodeCIDR(node, oldNode)
```

Comment:
this should be called only inside the block `if needUpdate` https://github.com/kubernetes/cloud-provider-gcp/pull/512/commits/14ad25f8ae106a4d2ddd00b87fe7a7b6a03c696d?diff=split&w=0#r1184347118

actually, it seems that encapsulating it in the function is not adding much value and you can leave it as it was before

Reply:
I have `reflect` comparing old with new. next step would be to remove the `needUpdate` altogether.

Reply:
got it now 👍